template_feed/Microsoft.DotNet.Common.ItemTemplates/content/ViewImports/.template.config/template.json

```diff
@@ -0,0 +1,27 @@
+{
+  "author": "Microsoft",
+  "classifications": [
+    "Web",
+    "ASP.NET"
+  ],
+  "name": "MVC ViewImports",
+  "generatorVersions": "[1.0.0.0-*)",
+  "groupIdentity": "Microsoft.AspNetCore.Mvc.ViewImports",
+  "precedence": "100",
+  "identity": "Microsoft.AspNetCore.Mvc.ViewImports",
+  "shortName": "viewimports",
+  "sourceName": "_ViewImports",
+  "primaryOutputs": [
+    {
+      "path": "_ViewImports.cshtml"
+    }
+  ],
+  "defaultName": "_ViewImports",
```

Comment:
Ok, if you set the sourceName to some string that doesn't appear anywhere in the template file names or content, it won't have any effect, maybe something like "xxx".

Also, we chatted a bit about the defaultName in templates. That is the final fallback for the "name".
The name resolution is:
1) The vale of the --name input (if it has a value)
2) The directory the template is being made in, which is the last directory in the -o path, or the current directory if no -o flag is given.
3) If #2 doesn't have a value (which can pretty much only happen if creating a template  in the *nix root directory '/'), then the defaultName is used.